File: src/main/java/org/dxworks/gitsecond/data/CommitData.java
package org.dxworks.gitsecond.data;

import lombok.*;

import java.util.Date;
import java.util.List;

@Data
@Builder
@EqualsAndHashCode
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PACKAGE)
public class CommitData {

    private String id;
    private String message;
    private String authorName;
    private String authorEmail;
    private Date date;

    private List<String> parentIds;

    private List<ChangesData> changeSets;

    public boolean isMergeCommit() {
        return parentIds != null && parentIds.size() >= 2;
    }
}

File: src/main/java/org/dxworks/gitsecond/data/CommitData.kt
package org.dxworks.gitsecond.data

import java.util.*

data class CommitData(var id: String, var message: String, var authorName: String, var authorEmail: String, var date: Date, var parentIds: List<String>, var changeSets: List<ChangesData>) {

    val isMergeCommit: Boolean
        get() = parentIds.size >= 2
}
